bundles/org.openhab.binding.metofficedatahub/src/main/java/org/openhab/binding/metofficedatahub/internal/MetOfficeDataHubSiteApiHandler.java

```diff
@@ -0,0 +1,890 @@
+            if (job != null) {
+                job.cancel(allowInterrupt);
+                dailyScheduler = null;
+            }
+        }
+    }
+
+    /**
+     * This handles the calculation of the delay until the next hourly forecast data should be retrieved.
+     * This calculates using a multiple since midnight to calculate when the next poll should occur.
+     * If the next poll is past midnight it is not scheduled as the daily synchronisation should reschedule a poll
+     * at that time.
+     */
+    private void scheduleNextHourlyForecastPoll() {
+        long millisSinceDayStart = DAY_IN_MILLIS - (DAY_IN_MILLIS - getMillisSinceDayStart());
+        long pollRateMillis = config.hourlyForecastPollRate * 3600000L;
+        long initialDelayTimeToFirstCycle = ((millisSinceDayStart - (millisSinceDayStart % pollRateMillis))
+                + pollRateMillis) - millisSinceDayStart;
+        initialDelayTimeToFirstCycle += RANDOM_GENERATOR.nextInt(60000);
+
+        if (initialDelayTimeToFirstCycle + millisSinceDayStart >= DAY_IN_MILLIS) {
+            logger.debug("Not scheduling poll after next daily cycle reset");
+        } else {
+            logger.debug("Can scheduling next Hourly forecast data poll to be in {} milliseconds at {}",
+                    initialDelayTimeToFirstCycle,
+                    millisToLocalDateTime(System.currentTimeMillis() + initialDelayTimeToFirstCycle));
+
+            // Schedule the first poll to occur after the given delay
+            hourlyForecastJob.scheduleExecution(initialDelayTimeToFirstCycle, () -> {
+                pollHourlyForecast();
+                scheduleNextHourlyForecastPoll();
+            });
+        }
+    }
+
+    /**
+     * This handles the calculation of the delay until the next hourly forecast data should be retrieved.
+     * This calculates using a multiple since midnight to calculate when the next poll should occur.
+     * If the next poll is past midnight it is not scheduled as the daily synchronisation should reschedule a poll
+     * at that time.
+     */
+    private void scheduleNextDailyForecastPoll() {
+        long millisSinceDayStart = DAY_IN_MILLIS - (DAY_IN_MILLIS - getMillisSinceDayStart());
+        long pollRateMillis = config.dailyForecastPollRate * 3600000L;
+        long initialDelayTimeToFirstCycle = ((millisSinceDayStart - (millisSinceDayStart % pollRateMillis))
+                + pollRateMillis) - millisSinceDayStart;
+        if (initialDelayTimeToFirstCycle + millisSinceDayStart > DAY_IN_MILLIS) {
+            logger.debug("Not scheduling poll after next daily cycle reset");
+        } else {
+            logger.debug("Can scheduling next Daily forecast data poll to be in {} milliseconds at {}",
+                    initialDelayTimeToFirstCycle,
+                    millisToLocalDateTime(System.currentTimeMillis() + initialDelayTimeToFirstCycle));
+
+            initialDelayTimeToFirstCycle += RANDOM_GENERATOR.nextInt(60000);
```

Comment:
Yeah :) I didn't want them to get annoyed at suddenly burst's on the hour from openHAB clients, and then get defensive on the API's. We suffer where I work with huge influxes, so this will at least not cause I expect a issue for the met office. That's the only real reason for it, and users won't notice this random delay, given its a once per hour refresh.